### PR TITLE
Multiclass support

### PR DESCRIPTION
**Pull request by timsetsfire**
OK.  

main add 
-- SoftmaxCrossEntropyWithLogits => for mutliclass problems.  Softmax calc'd internally in numerically stable manner.  
-- CrossEntropyWithLogits => more stable that CrossEntropy

ancillary add 
-- Identity activation for use with new loss functions.  

required
-- update to the main network class to check if the loss function is SoftmaxCrossEntropyWithLogits

## Cross Entropy with Logits, Spiral Dataset
```
from scratchnet.networks import Network
from scratchnet.activations import *
from scratchnet.losses import * 
from scratchnet.datasets import make_spiral

X, y = make_spiral(n_samples=600, n_features=2, n_classes=2)

## specify network
model = Network(n_features=2)
model.add_layer(16, Tanh(), random_state=42)
model.add_layer(8, Tanh(), random_state=42)
model.add_layer(1, Identity(), random_state=42)

## train network
history = model.train(X, y, loss=CrossEntropyWithLogits(), epochs=5000)

## make prediction
yhat = model.predict(X)
 
## assess accuracy
print("accuracy: {:.2f}".format(((1 / (1 + np.exp(-yhat)) > 0.5).squeeze() == y).mean()))
```
output 
`accuracy: 0.99`

## Softmax Cross Entropy with Logits, , Spiral Dataset
```
try:
    del model, history, yhat
except:
    pass

def softmax(y):
    ym = y.max(1)[:, np.newaxis]
    z = np.exp(y-ym)
    return z / (z.sum(1))[:, np.newaxis]



model = Network(n_features=2)
model.add_layer(16, Tanh(), random_state=42)
model.add_layer(8, Tanh(), random_state=42)
model.add_layer(2, Identity(), random_state=42)

history = model.train(X, y, loss=SoftmaxCrossEntropyWithLogits(), epochs=5000)

yhat = model.predict(X)

yhat = softmax(yhat) 

(np.argmax(yhat,1) == y).mean()

print("accuracy: {:.2f}".format((np.argmax(yhat,1) == y).mean()))
```
output
`accuracy: 0.99`

## Softmax Cross Entropy with Logits, Sklearn Digits dataset
```
try:
    del model, history, yhat, X, y
except:
    pass

from sklearn.datasets import load_digits
X,y = load_digits(return_X_y = True)
_, n_features = X.shape

model = Network(n_features=n_features)
model.add_layer(36, Tanh(), random_state=42)
model.add_layer(18, Tanh(), random_state=42)
model.add_layer(10, Identity(), random_state=42)

history = model.train(X, y, loss=SoftmaxCrossEntropyWithLogits(), epochs=500, learning_rate = 0.1)

yhat = softmax(model.predict(X)) 

print("accuracy: {:.2f}".format((np.argmax(yhat,1) == y).mean()))
```
output `accuracy: 0.98`

### CODE REVIEW

**Comment by AlbertDeFusco** [comment]
Completely lost this one!
